packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -114,7 +114,18 @@ async def handle_exc(self, e: Exception, message: HumanChatMessage):
         implementation is provided, however chat handlers (subclasses) should
         implement this method to provide a more helpful error response.
         """
-        await self._default_handle_exc(e, message)
+        self.log.error(e)
+        if self.config_manager.lm_provider.is_api_key_exc(e):
+            self.handle_api_key_exc(e, message)
+        else:
+            await self._default_handle_exc(e, message)
+
+    def handle_api_key_exc(self, e: Exception, message: HumanChatMessage):
+        provider_name = ""
+        if hasattr(self.config_manager, "lm_provider"):
+            provider_name = getattr(self.config_manager.lm_provider, "name", "")
+        response = f"Oops! There's a problem connecting to {provider_name}. Please update your {provider_name} API key in the chat settings."
+        self.reply(response, message)
```

Comment:
I don't think this method needs to exist on its own, since we can validate `self.config_manager.lm_provider` very quickly:

```py
def _default_handle_exc(self, e: Exception, message: HumanChatMessage):
    ...
    lm_provider = self.config_manager.lm_provider
    if lm_provider and lm_provider.is_api_key_exc(e):
        # <<< call self.reply(...) here
        return
    ...
```